docs/ADD_NEW_MIB

```diff
@@ -0,0 +1,53 @@
+#   ########################################################################
+#   Copyright 2021 Splunk Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#   ########################################################################
+
+We want to document here how to add a new MIB file to our server.
+
+1. create a new empty folder (like mkdir -p "$HOME/add_new_mib" && cd "$HOME/add_new_mib")
+2. curl https://raw.githubusercontent.com/etingof/pysmi/master/scripts/mibdump.py -o mibdump.py && chmod a+x mibdump.py
+3. Now we need to tell mibdump.py where are the new missing MIB files we want to import. You can do this by
+   specifying an additional --mib-source parameter with a custom absolute path. In this case, just for semplicity,
+   I have copied the MIB file I want to add to our MIB server into the current folder. That's why I have added
+   "--mib-source .". As you can see, we need also to tell mibdump.py where to find all the required imports each MIB
+   file might have. For example TintriMIB.txt needs these external symbols:
+   IMPORTS
+    MODULE-IDENTITY, OBJECT-TYPE, OBJECT-IDENTITY, NOTIFICATION-TYPE,
+    Integer32, Unsigned32, enterprises
+        FROM SNMPv2-SMI
+    DisplayString, DateAndTime
+        FROM SNMPv2-TC
+    MODULE-COMPLIANCE, OBJECT-GROUP, NOTIFICATION-GROUP
+        FROM SNMPv2-CONF;
+    and these symbols are defined in SNMPv2-SMI, SNMPv2-TC, and SNMPv2-CONF.
+
+4. run mibdump.py --mib-source http://mibs.thola.io/asn1/@mib@ \
+                  --mib-source . --destination-format pysnmp \
+                  --destination-directory=./temp <your-mib-file>
+4.1. If for some reason your environment doesn't have access to Internet (and hence cannot connect to
+     http://mibs.thola.io/asn1/@mib@) you could simply use all the original MIB files we store in the
+     splunk-connect-for-snmp-mib-server project under the folder $ROOT_PROJECT/mibs/asn1. Something like this
+     would work as well (assuming on extracted "splunk-connect-for-snmp-mib-server" project under
+     ""$HOME/PycharmProjects/splunk-connect-for-snmp-mib-server").
+     mibdump.py --mib-source "$HOME/PycharmProjects/splunk-connect-for-snmp-mib-server/mibs/asn1" \
+                --mib-source . --destination-format pysnmp --destination-directory=./temp3 TintriMIB.txt
+
+5. Copy <your-mib-file> to $ROOT_PROJECT/mibs/asn1
```

Comment:
I think we should make sure if "your-mib-file" is named as the MIB name correctly before copying it to $ROOT_PROJECT/mibs/asn1. If it was not named as its MIB name, we need to rename it as its MIB name and copy the "renamed-mib-file" to $ROOT_PROJECT/mibs/asn1

**For example**, in this case, we should

1. rename "TintriMIB.txt" to "VMSTORE-MIB"(where VMSTORE-MIB is its MIB name, which resides at the 1st line inside itself)
2. copy "VMSTORE-MIB" into $ROOT_PROJECT/mibs/asn1.

The reason why we need to have the raw mib files named as the MIB name is when users try to query by the mib-string (on the poller side), they will specify it in this syntax 
 

> Syntax: [ "MIB-Files", "MIB object name" "MIB index number"]

Where "MIB-Files" must be the correct MIB name, e.g. 

> - ["VMSTORE-MIB", "vmstoreHostname"]

The "MIB-Files" here ("VMSTORE-MIB") will be used to search in both $ROOT_PROJECT/mibs/asn1 and $PROJECT_ROOT/mibs/pysnmp. If we just copy "TintriMIB.txt" to ROOT_PROJECT/mibs/asn1 directly, it would throw a 404 error like 
- "GET /files/asn1/VMSTORE-MIB HTTP/1.1" 404

Reply:
Excellent catch! I wasn't really aware of this, I just assumed we had the correct file name. I have fixed them both:
- the documentation, and
- the 2 newly-added mib files.

Thanks a lot!

Reply:
Thanks for the changes! :)